Mathlib/CategoryTheory/Filtered/Final.lean

```diff
@@ -288,6 +289,32 @@ instance [IsCofiltered C] (X : C × C) : IsCofiltered (CostructuredArrow (diag C
 instance Functor.diag_initial_of_isFiltered [IsCofiltered C] : Initial (Functor.diag C) :=
   initial_of_isCofiltered_costructuredArrow _
 
+/-- The functor `StructuredArrow.proj : StructuredArrow Y T ⥤ C` is final if `T : C ⥤ D` is final
+and `C` is filtered. -/
+instance StructuredArrow.proj_final_of_filtered [IsFiltered C]
+    {D : Type u₂} [Category.{v₁} D] (T : C ⥤ D) [Final T] (Y : D) :
+    Final (StructuredArrow.proj Y T) := by
+  haveI : ∀ (X : C), IsFiltered (StructuredArrow X (proj Y T)) := fun X => by
+    haveI : IsFiltered (StructuredArrow Y (Under.forget X ⋙ T)) := by
+      revert Y
+      rw [← Functor.final_iff_isFiltered_structuredArrow]
+      exact final_comp (Under.forget X) T
+    apply IsFiltered.of_equivalence (ofStructuredArrowProjEquivalence T Y X).symm
+  apply final_of_isFiltered_structuredArrow
```

Comment:
It turns out that we can generalize universes here by going through `IsConnected` instead of `IsFiltered`. The following proof works without constraining the universes of the morphisms of `D`:
```lean
  refine ⟨fun X => ?_⟩
  rw [isConnected_iff_of_equivalence (ofStructuredArrowProjEquivalence T Y X)]
  exact (final_comp (Under.forget X) T).out 
```
So I think this theorem should be moved out of the `LocallySmall` section.

It might be possible that this can be done for other theorems in this section, I haven't checked.